### ISSUE TITLE
[Units] [Traits] Allow Traits to have mutually exclusive attribute

### ISSUE THREAD

**Issue by knyghtmare** [wesnoth/wesnoth#6598]
### Describe the desired feature

Example Situation Case:

I have a unit which has access to 4 traits on being recruited: A, B, C, and D. Usually, on recruitment, the unit can have...
AB, BC, CD, AD, AC and BD combinations of traits. 

> Mutually exclusive is a statistical term describing two or more events that cannot happen simultaneously. It is commonly used to describe a situation where the occurrence of one outcome supersedes the other.

The feature I want is to make 2 traits mutually exclusive. Such as A and B.
Thus, in that case, if recruited the Unit can have trait combinations of BC, CD, AD, AC and BD...but never AB.

An example would be a unit which can have both weak and strong traits. In that case, a unit with strong and weak traits would be strange to see.

Before filing this FR, I read https://wiki.wesnoth.org/UnitsWML#.5Btrait.5D but found that `[trait][/trait]` had no way to make traits mutually exclusive.

Example:
```ini
    [trait]
        id=intelligent

        # no recruited/generated unit will have intelligent, dim as traits
        mutually_exclusive=dim

        [effect]
            apply_to=max_experience
            increase=-20%
        [/effect]
    [/trait]

    [trait]
        id=dim

        # no recruited/generated unit will have intelligent, dim as traits
        mutually_exclusive=intelligent


        [effect]
            apply_to=max_experience
            increase=-20%
        [/effect]
    [/trait]
```

**Comment by CelticMinstrel** [comment]
I feel like… I thought this already existed? I guess I might've been thinking of the similar feature for `[advancement]`… or maybe there was a pull request that didn't get merged…

**Comment by knyghtmare** [comment]
> I feel like… I thought this already existed? I guess I might've been thinking of the similar feature for `[advancement]`… or maybe there was a pull request that didn't get merged…

Advancements are mutually exclusive but not traits, yes.

**Comment by Kingofd** [comment]
I will see what I can do here

**Comment by CelticMinstrel** [comment]
Rather than your example, it might be better to use the same syntax that `[advancement]` uses for this feature.

**Comment by Kingofd** [comment]
Should be closed, was done in #7109, already merged
